### PR TITLE
Use get_identifier() instead of get(ID_identifier) on symbols

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

**Comment by tautschnig** [comment]
The changes are probably good, but it's really hard to review: The commit messages and the actual changes seem largely unrelated, and few of the latter actually relate to the subject of this pull request.

File: src/util/std_types.h [hunk 1]
```diff
@@ -130,6 +133,52 @@ class symbol_typet:public typet
   }
 };
 
+/*! \brief Cast a generic typet to a \ref tag_typet
```
Comment (peterschrammel) [inline src/util/std_types.h]: We should converge towards using `/// comments` for doxygen.

**Review by peterschrammel — APPROVED** [review]

**Review by tautschnig — CHANGES_REQUESTED** [review]
Echoing what I've said before as it got assigned to me: "The commit messages and the actual changes seem largely unrelated, and few of the latter actually relate to the subject of this pull request."

**Comment by kroening** [comment]
This should now be straight-forward to review.

**Review by tautschnig — APPROVED** [review]

**Pull request merged**